### PR TITLE
Refactor how images are built

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | New way to generate docker images and aliases. 
| Type?         | refacto
| BC breaks?    | yes
| Deprecations? | no
| How to test?  | 

## How to manage versions

The new way allow anyone to add 
```python
VERSIONS = {
    '1.7.6.8': (
        '5.6',
        '7.1',
        '7.2',
    ),
    'nightly': (
        '7.1',
        '7.2',
        '7.3',
        '7.4',
    ),
}
```

This will create these `tag: path`:
```json
{
    "nightly-7.1-apache": "/home/got/projects/prestashop/docker/images/nightly/7.1-apache",
    "nightly-7.4-fpm": "/home/got/projects/prestashop/docker/images/nightly/7.4-fpm",
    "nightly-7.4-apache": "/home/got/projects/prestashop/docker/images/nightly/7.4-apache",
    "nightly-7.2-apache": "/home/got/projects/prestashop/docker/images/nightly/7.2-apache",
    "nightly-7.1-fpm": "/home/got/projects/prestashop/docker/images/nightly/7.1-fpm",
    "nightly-7.3-apache": "/home/got/projects/prestashop/docker/images/nightly/7.3-apache",
    "nightly-7.2-fpm": "/home/got/projects/prestashop/docker/images/nightly/7.2-fpm",
    "nightly-7.3-fpm": "/home/got/projects/prestashop/docker/images/nightly/7.3-fpm",
    "1.7.6.8-5.6-fpm": "/path/to/prestashop/docker/images/1.7.6.8/5.6-fpm",
    "1.7.6.8-7.2-fpm": "/path/to/prestashop/docker/images/1.7.6.8/7.2-fpm",
    "1.7.6.8-7.1-fpm": "/path/to/prestashop/docker/images/1.7.6.8/7.1-fpm",
    "1.7.6.8-7.2-apache": "/path/to/prestashop/docker/images/1.7.6.8/7.2-apache",
    "1.7.6.8-7.1-apache": "/path/to/prestashop/docker/images/1.7.6.8/7.1-apache",
    "1.7.6.8-5.6-apache": "/path/to/prestashop/docker/images/1.7.6.8/5.6-apache"
}
```

With these aliases: 
```
Aliases for 1.7.6.8-5.6-apache
	1.7-5.6
	1.7.6-5.6
	1.7.6.8-5.6
Aliases for 1.7.6.8-7.1-apache
	1.7-7.1
	1.7.6-7.1
	1.7.6.8-7.1
Aliases for 1.7.6.8-7.2-fpm
	1.7-fpm
	1.7.6-fpm
	1.7.6.8-fpm
Aliases for 1.7.6.8-7.2-apache
	1.7-7.2
	1.7
	1.7-apache
	1.7.6-7.2
	1.7.6
	1.7.6-apache
	1.7.6.8-7.2
	1.7.6.8
	1.7.6.8-apache
	latest
Aliases for nightly-7.1-apache
	nightly-7.1
Aliases for nightly-7.2-apache
	nightly-7.2
Aliases for nightly-7.3-apache
	nightly-7.3
Aliases for nightly-7.4-fpm
	nightly-fpm
Aliases for nightly-7.4-apache
	nightly-7.4
	nightly
	nightly-apache
```

### CODE REVIEW

File: prestashop_docker/docker_api.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            # Something went wrong, retry
+            time.sleep(self.sleep_time)
+            DockerApi.retries += 1
+            if DockerApi.retries >= 10:
+                raise requests.HTTPError(resp.text)
+
+            return self.execute(request_url)
+        else:
+            DockerApi.retries = 0
+            # Data not in cache, we must wait because of GitHub API rate limits
```
Comment (matks) [inline prestashop_docker/docker_api.py:77]: 🤔 I dont understand the relation between DockerApi and GitHub rate limit ?

File: prestashop_docker/tag_manager.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        tags = self.docker_api.get_tags()
+        for tag in tags:
+            if tag['name'] == version:
+                return True
+
+        return False
+
+    def get_versions(self):
+        '''
+        Test if a version is already on Docker Hub
```
Comment (matks) [inline prestashop_docker/tag_manager.py:96]: Inaccurate 😉 probably copy-paste

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,21 @@
+[nosetests]
+verbosity=3
+with-doctest=1
+detailed-errors=1
+with-coverage=1
+cover-html=1
+cover-tests=1
+cover-erase=1
+cover-package=core_weekly
```
Comment (matks) [inline setup.cfg:9]: I see you 👀 copying my code and stuff

Comment (PierreRambaud) [reply setup.cfg:9]: That was my code, it was late :sweat_smile:

Comment (matks) [reply setup.cfg:9]: I know my friend ❤️ I'm just trolling :trollface:

**Review by Progi1984 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]
Not a Python expert but seems good to me

File: Dockerfile-nightly.model [hunk 4]
```diff
@@ -1,12 +1,12 @@
 FROM google/cloud-sdk:latest as nightly-storage
 
-RUN gsutil cp `gsutil ls gs://prestashop-core-nightly/ | grep -E 'develop.+\.zip$' | tail -1` /tmp/prestashop.zip
+RUN gsutil cp `gsutil ls gs://prestashop-core-nightly/ | grep -E 'develop.+\.zip$$' | tail -1` /tmp/prestashop.zip
```
Comment (matks) [inline Dockerfile-nightly.model:3]: Why this double dollar sign?

**Review by matks — APPROVED** [review]
Looks OK for me, I just have one question (see above)

File: Dockerfile-nightly.model (hunk 4 above)
Comment (PierreRambaud) [reply Dockerfile-nightly.model:3]: Because it's a template and $ is for variables.

**Comment by PierreRambaud** [comment]
Code ok, still need README + Unit tests :)

**Review by Progi1984 — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Tell me for readme and How to use file :)

**Review by Progi1984 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by atomiix** [comment]
The readme and how to use seem good to me 👍

**Comment by Progi1984** [comment]
Thanks @PierreRambaud

**Pull request merged**